src/adapter/src/coord/ddl.rs

```diff
@@ -762,6 +768,36 @@ impl Coordinator {
         self.update_compute_base_read_policies(compute_policies);
     }
 
+    fn update_default_retention(&mut self) {
+        let lag = self.catalog().system_config().default_retention_timestamp();
+        let policy = ReadPolicy::lag_writes_by(lag, SINCE_GRANULARITY);
+        let storage_policies = self
+            .catalog()
+            .entries()
+            .filter(|entry| {
+                entry.item().uses_default_retention()
+                    && entry.item().is_compute_object_on_cluster().is_none()
+            })
+            .map(|entry| (entry.id(), policy.clone()))
+            .collect::<Vec<_>>();
+        let compute_policies = self
+            .catalog()
+            .entries()
+            .filter_map(|entry| {
+                if let (true, Some(cluster_id)) = (
+                    entry.item().uses_default_retention(),
+                    entry.item().is_compute_object_on_cluster(),
+                ) {
+                    Some((cluster_id, entry.id(), policy.clone()))
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
```

Comment:
It used to be the case that materialized views and introspection sources had both a storage object and a compute object, with the same GlobalId, and we needed to update the read policies of both separately.

If that's still the case then this might not work? Though it's possible that's no longer true.

Reply:
Changed to what we discussed over slack: iterating on storage/compute_read_capabilities directly

Reply:
It looks like that should work, but CI is not happy about something.